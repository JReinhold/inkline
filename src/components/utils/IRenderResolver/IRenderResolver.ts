import { defineComponent, h, PropType, RenderFunction } from 'vue';

export default defineComponent({
    props: {
        /**
         * The primitive or render function to render.
         * @type String | Number | Boolean | RenderFunction
         * @default
         * @name data
         */
        data: {
            type: [String, Number, Boolean, Function] as PropType<
                number | boolean | string | RenderFunction
            >,
            required: true
        },
        /**
         * The tag to use for rendering primitives
         * @type String
         * @default div
         * @name tag
         */
        tag: {
            type: String,
            default: 'div'
        }
    },
    setup(props) {
        return () => (typeof props.data === 'function' ? props.data() : h(props.tag, props.data));
    }
});
