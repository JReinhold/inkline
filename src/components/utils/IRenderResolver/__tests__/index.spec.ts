import { render } from '@testing-library/vue';
import { IRenderResolver } from '@inkline/inkline/components/utils/IRenderResolver';
import { h } from 'vue';

describe('Components', () => {
    describe('IRenderResolver', () => {
        it('should render component render function', () => {
            const props = {
                data: () => h('div', 'example')
            };

            const wrapper = render(IRenderResolver, {
                props
            });

            expect(wrapper.html()).toMatchSnapshot();
        });

        it('should render string', () => {
            const props = {
                data: 'example'
            };

            const wrapper = render(IRenderResolver, {
                props
            });

            expect(wrapper.html()).toMatchSnapshot();
        });

        it('should render string using specified tag', () => {
            const props = {
                data: 'example',
                tag: 'strong'
            };

            const wrapper = render(IRenderResolver, {
                props
            });

            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
