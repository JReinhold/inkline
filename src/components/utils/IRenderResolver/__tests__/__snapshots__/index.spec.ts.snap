// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components > IRenderResolver > should render component render function 1`] = `"<div>example</div>"`;

exports[`Components > IRenderResolver > should render string 1`] = `"<div>example</div>"`;

exports[`Components > IRenderResolver > should render string using specified tag 1`] = `"<strong>example</strong>"`;
